src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -101,7 +101,7 @@ public static unsafe OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Spa
                 }
 
                 ref sbyte decodingMap = ref MemoryMarshal.GetReference(DecodingMap);
-                srcMax = srcBytes + (uint)maxSrcLength;
+                srcMax = srcBytes + maxSrcLength;
```

Comment:
The `uint`-cast was there to avoid the `movsxd` (on x86), so removing the cast will introduce the `movsxd` and I expect that be cheaper than having a `if` to guard the loop.